test/coverage/datamodel/UuidSerializerTest.cpp

```diff
@@ -23,16 +27,235 @@ class TestFixture : public testing::Test {
+}
+
+// Test serialization with mixed case letters
+TEST_F(TestUuidSerializer, SerializeWithMixedCaseLetters) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x1234567890ABcDEF);
+	uuid.set_lsb(0x00dcbA0987654321);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value in lowercase
+	EXPECT_EQ(serialized_uuid, "12345678-90ab-cdef-00dc-ba0987654321");
+}
+
+// Test serialization with leading zeros and mixed case letters
+TEST_F(TestUuidSerializer, SerializeWithLeadingZerosAndMixedCaseLetters) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x00001234567890AB);
+	uuid.set_lsb(0xFedcba0987654982);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "00001234-5678-90ab-fedc-ba0987654982");
+}
+
+// Test serialization with leading/trailing zeros and mixed case letters
+TEST_F(TestUuidSerializer,
+       SerializeWithLeadingZerosAndTrailingZerosAndMixedCaseLetters) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x00001234567890AB);
+	uuid.set_lsb(0xFedcba0987600000);
+
+	std::string serialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	// Assert the serialized UUID matches the expected value
+	EXPECT_EQ(serialized_uuid, "00001234-5678-90ab-fedc-ba0987600000");
+}
+
+// Test string deserialization
+TEST(DeserializerTest, DeserializeUUID) {
+	// Define a UUID string in the traditional format
+	std::string uuid_str = "12345678-9abc-def0-fedc-ba9876543210";
+	// Deserialize the UUID string
+	uprotocol::v1::UUID deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::deserialize(uuid_str);
+	// Assert the deserialized UUID matches the expected values
+	EXPECT_EQ(deserialized_uuid.msb(), 0x123456789ABCDEF0);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0xFEDCBA9876543210);
+}
+
+// Test deserialization with leading/trailing zeros and mixed case letters
+TEST_F(TestUuidSerializer,
+       DeserializeWithLeadingZerosAndTrailingZerosAndMixedCaseLetters) {
+	std::string uuid_str = "00001234-5678-90ab-feDc-ba0987600000";
+
+	uprotocol::v1::UUID deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::deserialize(uuid_str);
+
+	// Assert the deserialized UUID matches the expected values
+	EXPECT_EQ(deserialized_uuid.msb(), 0x00001234567890aB);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0xFedcBA0987600000);
+}
+
+// Test invalid string deserialization
+TEST(DeserializerTest, InvalidUUIDFormat) {
+	// Define an invalid UUID string (missing dashes)
+	std::string invalid_uuid_str = "123456789abcdef0123456789abcdef0";
+	// Assert that deserialization throws an invalid argument exception
+	EXPECT_THROW(uprotocol::datamodel::serializer::uuid::AsString::deserialize(
+	                 invalid_uuid_str),
+	             std::invalid_argument);
```

Comment:
Hmmm - for the purpose of code coverage, shouldn't this be multiple tests, one for each dash position ?
x----
-x---
--x-- etc ?

Reply:
In term of coverage this should be good, but I'll add few more test for other dash placement